src/mame/toaplan/toaplan2.cpp

```diff
@@ -661,11 +666,11 @@ READ_LINE_MEMBER(toaplan2_state::c2map_r)
 	// bit 4 high signifies secondary CPU is ready
 	// bit 5 is tested low before V-Blank bit ???
 
-	return m_soundlatch->pending_r() ? 0x00 : 0x01;
+	return m_soundlatch[0]->pending_r() ? 0x00 : 0x01;
 }
 
 
-u16 toaplan2_state::ghox_p1_h_analog_r()
+u16 ghox_state::ghox_p1_h_analog_r()
 {
 	const s8 new_value = m_io_pad[0]->read();
 	if (new_value == m_old_p1_paddle_h) return 0;
```

Comment:
These two functions seem overly complex.  The subtraction will yield zero if the old value and new value are identical anyway so the extra `if` is unnecessary.  You could reduce it to:
```c++
u16 ghox_state::ghox_p1_h_analog_r()
{
	const s8 new_value = m_io_pad[0]->read();
	const s8 result = new_value - m_old_p1_paddle_h;
	if (!machine().side_effects_disabled())
		m_old_p1_paddle_h = new_value;
	return result;
}
```

Also, if you turned the old paddle values into an array, you could use a single function template for both cases and avoid the duplicated code.